db/bigtable.go

```diff
@@ -448,16 +514,44 @@ func (bigtable *Bigtable) SaveAttestationAssignments(epoch uint64, assignments m
 		validatorsPerSlot[attesterslot] = append(validatorsPerSlot[attesterslot], validator)
 	}
 
+	muts := make([]*gcp_bigtable.Mutation, 0, MAX_BATCH_MUTATIONS)
+	keys := make([]string, 0, MAX_BATCH_MUTATIONS)
+
 	for slot, validators := range validatorsPerSlot {
-		mut := gcp_bigtable.NewMutation()
+
 		for _, validator := range validators {
-			mut.Set(ATTESTATIONS_FAMILY, fmt.Sprintf("%d", validator), ts, []byte{})
+			mut := gcp_bigtable.NewMutation()
+			key := fmt.Sprintf("%s:%s:%s", bigtable.chainId, bigtable.validatorIndexToKey(validator), bigtable.reversedPaddedEpoch(epoch))
+			mut.Set(ATTESTATIONS_FAMILY, fmt.Sprintf("%d", slot), ts, []byte{})
+
+			muts = append(muts, mut)
+			keys = append(keys, key)
+		}
+
+		if len(muts) == MAX_BATCH_MUTATIONS {
```

Comment:
This works as long as validators attestation assignments per slot are less than MAX_BATCH_MUTATIONS. Otherwise muts array could be bigger than MAX_BATCH_MUTATIONS, not triggering the applybulk. Right now with 800k validators this is more threoretical but if we ever change MAX_BATCH_MUTATIONS or validator size groes by 4x we run into this problem.